app/services/foreman_salt/fact_parser.rb

```diff
@@ -89,6 +89,7 @@ def support_interfaces_parsing?
     def os_hash
       name = facts[:os]
       (_, major, minor, sub) = /(\d+)\.?(\d+)?\.?(\d+)?/.match(facts[:osrelease]).to_a
+      minor = minor.to_s if minor.nil?
```

Comment:
This doesn't really make sense? 
Set minor to "" (empty) if minor == nil? Do you really want to do that?

Reply:
```suggestion
      minor = "" if minor.nil?
```

is more readable

Reply:
It's more or less the same logic as for the puppet fact parser here https://github.com/theforeman/foreman/blob/9d93eceaf05d673513e3961df1f154f11c7bb179/app/services/puppet_fact_parser.rb#L10

I agree, this is more readable.